### DISCUSSION TITLE
Are real time remote config updates available?

### DISCUSSION THREAD

**Discussion by jeroenptrs** [invertase/react-native-firebase#7005] [Q&A] [answered]
In the changelog I noticed the versions needed for this https://firebase.google.com/docs/remote-config/propagate-updates-realtime?hl=en&authuser=0 are in the latest version. What I couldn't find is whether the actual functionality of real-time RC updates are supported, as I'm assuming some implementation work needs to be done for this.

Does anybody know whether this is already supported or if it's on the roadmap?

**Comment by mikehardy** [comment]
It is not supported though we do have the necessary underlying SDK versions wrapped + released now - we just haven't exposed the APIs

All of the features that the underlying SDKs have added in the last few versions are on the roadmap (in general, any SDK features should be + will be exposed over time...) but I cannot give an ETA

If this is important for your project, we welcome PRs and I'm happy to collaborate with you to get one merged in. Our tests app is how we do development - this doc shows how to check it out + get it set up + run it to make sure baseline is good https://github.com/invertase/react-native-firebase/blob/main/tests/README.md

After you've got a good baseline, my general process for wrapping new APIs is first to define it in the index.d.ts following however firebase-js-sdk does it, then add it to index.js with stubs that just console.log and add an e2e test at the same time to exercise the stubs + make sure everything is hooked up. Then I get to work in the obj-c + java with stubs and make index.js call the native stubs checking native logs. Then finally I call the actual underlying firebase native SDK methods. That's the full description and sometimes it's just a few minutes for each step - it can go pretty quickly.

**Comment by alex-mironov** [comment]
@mikehardy do you know by chance if this feature request is still on the roadmap or has somebody picked it up?

**Comment by hstaudacher** [comment]
This would be an awesome addition and really helpful.

**Comment by mikehardy** [comment, marked as answer]
Just a note that realtime remote-config updates *are* out now. There were a couple of issues in the underlying firebase-ios-sdk that were discovered during our testing here, but they do work now as is, and all known issues will be fixed with firebase-ios-sdk 10.12.0 when it comes out soon
